@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -562,89 +520,258 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
+                ? fillTechnique.lineWidth || 0.0
+                : 0.0;
+            const hasEdges =
+                edgeWidth > 0.0 ||
+                isSolidLineTechnique(technique) ||
+                isDashedLineTechnique(technique) ||
+                isLineTechnique(technique);
+
+            for (const polygon of geometry) {
+                const rings: Ring[] = [];
+                for (const outline of polygon.rings) {
+                    const ringContour: number[] = [];
+                    let ringEdges: boolean[] | undefined;
+                    let ringTexCoords: number[] | undefined;
+                    for (let coordIdx = 0; coordIdx < outline.coordinates.length; ++coordIdx) {
+                        const geoPoint = outline.coordinates[coordIdx];
+                        const { x, y } = projection.projectPoint(geoPoint, worldPos).sub(center);
+                        ringContour.push(x, y);
+
+                        if (hasEdges && outline.outlines !== undefined) {
+                            const edge = outline.outlines[coordIdx];
+                            if (ringEdges === undefined) {
+                                ringEdges = [edge];
+                            } else {
+                                ringEdges.push(edge);
+                            }
                         }
-                    }
 
-                    if (isTextured) {
-                        const { x: u, y: v } = texCoordProjection
-                            .projectPoint(geoPoint, texCoord)
-                            .sub(texCoordBox.min)
-                            .divide(texCoordBoxSize);
-                        if (ringTexCoords === undefined) {
-                            ringTexCoords = [u, v];
-                        } else {
-                            ringTexCoords.push(u, v);
+                        if (isTextured) {
+                            const { x: u, y: v } = texCoordProjection
+                                .projectPoint(geoPoint, texCoord)
+                                .sub(texCoordBox.min)
+                                .divide(texCoordBoxSize);
+                            if (ringTexCoords === undefined) {
+                                ringTexCoords = [u, v];
+                            } else {
+                                ringTexCoords.push(u, v);
+                            }
                         }
                     }
+                    rings.push(new Ring(ringContour, ringEdges, ringTexCoords));
                 }
-                rings.push(new Ring(ringContour, ringEdges, ringTexCoords));
+                polygons.push(rings);
             }
-            polygons.push(rings);
+
+            if (technique.name === "fill" || technique.name === "extruded-polygon") {
+                this.applyPolygonTechnique(polygons, technique, techniqueIndex, featureId, env);
+            } else if (technique.name === "line") {
+                polygons.forEach(rings => {
```

Comment:
Same as above. A lot of duplicated code that breaks readability. Would be nice to simplify.